Lib/fontTools/ttLib/ttGlyphSet.py

```diff
@@ -178,16 +188,47 @@ def _setCoordinates(glyph, coord, glyfTable):
 
 
 class _TTVarGlyph(_TTGlyph):
-	def __init__(self, ttFont, glyphName, location):
-		self._ttFont = ttFont
+	def __init__(self, glyphSet, glyphName, location):
+
+		super().__init__(glyphSet._glyphs, glyphSet._glyphs[glyphName])
+
+		self._glyphSet = glyphSet
+		self._ttFont = glyphSet._ttFont
+		self._glyphs = glyphSet._glyphs
 		self._glyphName = glyphName
 		self._location = location
-		# draw() fills these in
-		self.width = self.height = self.lsb = self.tsb = None
 
 
-class _TTVarGlyphGlyf(_TTVarGlyph):
+class _TTVarGlyphCFF(_TTVarGlyph):
+	def draw(self, pen):
+		varStore = self._glyphs.varStore
+		if varStore is None:
+			blender = None
+		else:
+			from fontTools.varLib.varStore import VarStoreInstancer
+
+			vsInstancer = getattr(self._glyphSet, "vsInstancer", None)
+			if vsInstancer is None:
+				self._glyphSet.vsInstancer = vsInstancer = VarStoreInstancer(
+					varStore.otVarStore, self._ttFont["fvar"].axes, self._location
+				)
+			blender = vsInstancer.interpolateFromDeltas
+		self._glyph.draw(pen, blender)
+
+		self.width = self._ttFont["hmtx"][self._glyphName][0]
+		if "HVAR" in self._ttFont:
+			hvar = self._ttFont["HVAR"].table
+			varidx = self._ttFont.getGlyphID(self._glyphName)
+			if hvar.AdvWidthMap is not None:
+				varidx = hvar.AdvWidthMap.mapping[self._glyphName]
+			vsInstancer = VarStoreInstancer(
+				hvar.VarStore, self._ttFont["fvar"].axes, self._location
+			)
+			delta = vsInstancer[varidx]
+			self.width += delta
```

Comment:
if you draw the same glyph multiple times, wouldn't this increment its advance width in-place all the time?

Reply:
Yes, but this should be resolved in #2789

Reply:
I don't think so because `self.width` was reset a few lines earlier.